### ISSUE TITLE
Implicit assumption of spin ('up', 'do') order in pytriqs.operators.util.observables.S_op

### ISSUE THREAD

**Issue by HugoStrand** [TRIQS/triqs#536]
We hit a hidden assumption in the spin operators when moving the block Green's function block order from lexicographic to the order provided by the user in a gf_struct list.

The spin operators do not care about spin labels and instead assumes that spin-up always is the first spin index. E.g. if the spin labels are reversed the S_z operator suddenly changes sign.

This in combination with the block order change is a problem for users that want to compare with their previous results, since the spin block order is reversed.

[The lexicographic order always gives GfBlocks with block order ('do', 'up')].

**Comment by Wentzell** [comment]
With PR #559 the notion of spin (i.e. block) order more sense.

TODO: Update the documentation of S_op to include the implicit assumption that first spin is 'up'.

**Comment by Wentzell** [comment]
The documentation is in fact already very explicit about what it does. It is not necessarily assumed that the first spin is 'up' but the order is rather given in the ```spin_names``` tuple.

Cf. [S_op-Documentation](https://triqs.github.io/triqs/master/reference/operators/util/observables.html?highlight=s_op#pytriqs.operators.util.observables.S_op)

**Comment by HugoStrand** [comment]
I disagree, `S_op` assumes that the first index in `spin_names` has a spin-z projection that is 1/2. 
This causes the spin operators to be wrong when called with `spin_names=['down', 'up']`.

Here is a minimal example:
```python
from pytriqs.operators.util import S_op
Sz = S_op('z', spin_names=['up', 'down'], orb_names=[0], off_diag=False)
print 'Sz (correct) =', Sz
Sz = S_op('z', spin_names=['down', 'up'], orb_names=[0], off_diag=False)
print 'Sz (wrong) =', Sz
```
which gives
```
Sz (correct) = -0.5*c_dag('down_0',0)*c('down_0',0) + 0.5*c_dag('up_0',0)*c('up_0',0)
Sz (wrong) = 0.5*c_dag('down_0',0)*c('down_0',0) + -0.5*c_dag('up_0',0)*c('up_0',0)
```
Does this make the problem more clear?
